dockerfiles/theia/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM node:8-alpine
+# build dependencies requireed to compile a custom Theia
+RUN apk add --no-cache make gcc g++ python git openssh bash
+WORKDIR /home/theia
+# build Theia with all extensions to persist yarn cache in the image and
+# have default Theia build in the workspace in case no plugins are requested
+ADD https://raw.githubusercontent.com/theia-ide/theia-apps/master/theia-full-docker/latest.package.json /home/theia/package.json
+ADD theia-default-package.json /home/default/theia/package.json
+ADD src/main.js /theia_launcher/theia_launcher.js
+RUN yarn && \
+    yarn theia build && \
+    rm -rf * && \
+    cd /home/default/theia && \
+    yarn && \
+    yarn theia build
+EXPOSE 3000
+ENV USE_LOCAL_GIT=true \
+    GITHUB_TOKEN=957257352ad57b7853c8b8714d2201bf4d337b5e
+#    ENV THEIA_PLUGINS=[\"@theia\/extension-manager\",\"\@theia\/python\"]
```

Comment:
should we remove this comment ?

Reply:
we leave it for testing purpose, but we also can use  `-e THEIA_PLUGINS=...` for this, so I remove it

Reply:
@evidolob you may add a README or in the header of Dockerfile on how to use that image